rs/bitcoin/kyt/tests/tests.rs

```diff
@@ -162,14 +247,86 @@ fn test_get_inputs() {
-            );
+            let response = Decode!(bytes, CheckTransactionResult).unwrap();
+            assert!(matches!(response, Ok(CheckTransactionResponse::Passed)));
         }
         WasmResult::Reject(msg) => panic!("unexpected reject: {}", msg),
     }
+
+    let cycles_after = env.cycle_balance(setup.universal_canister);
+    let expected_cost =
+        CHECK_TRANSACTION_CYCLES_SERVICE_FEE + 2 * get_tx_cycle_cost(INITIAL_MAX_RESPONSE_BYTES);
+    let actual_cost = cycles_before - cycles_after;
+    assert!(actual_cost > expected_cost);
+    assert!(actual_cost - expected_cost < UNIVERSAL_CANISTER_CYCLE_MARGIN);
+}
+
+#[test]
+fn test_check_transaction_error() {
+    let setup = setup_env();
+    let cycles_before = setup.env.cycle_balance(setup.universal_canister);
+    let mut txid =
+        Txid::from_str("a80763842edc9a697a2114517cf0c138c5403a761ef63cfad1fa6993fa3475ed")
+            .unwrap()
+            .as_ref()
+            .to_vec();
+
+    // Test for cycles not enough
+    let call_id = submit_universal_call(
+        &setup,
+        setup.kyt_canister,
+        "check_transaction",
+        Encode!(&CheckTransactionArgs { txid: txid.clone() }).unwrap(),
+        CHECK_TRANSACTION_CYCLES_REQUIRED - 1,
+    )
+    .expect("submit_call failed to return call id");
+    let result = setup
+        .env
+        .await_call(call_id)
+        .expect("the fetch request didn't finish");
+    match &result {
+        WasmResult::Reply(bytes) => {
+            let response = Decode!(bytes, CheckTransactionResult).unwrap();
+            assert!(matches!(
+                response,
+                Ok(CheckTransactionResponse::NotEnoughCycles { .. })
+            ));
+        }
+        WasmResult::Reject(msg) => panic!("unexpected reject: {}", msg),
+    };
+    let cycles_after = setup.env.cycle_balance(setup.universal_canister);
+    let expected_cost = CHECK_TRANSACTION_CYCLES_SERVICE_FEE;
+    let actual_cost = cycles_before - cycles_after;
+    assert!(actual_cost > expected_cost);
+    assert!(actual_cost - expected_cost < UNIVERSAL_CANISTER_CYCLE_MARGIN);
+
+    // Test for malformatted txid
+    let cycles_before = setup.env.cycle_balance(setup.universal_canister);
+    txid.pop();
+    let call_id = submit_universal_call(
+        &setup,
+        setup.kyt_canister,
+        "check_transaction",
+        Encode!(&CheckTransactionArgs { txid }).unwrap(),
+        CHECK_TRANSACTION_CYCLES_REQUIRED,
+    )
+    .expect("submit_call failed to return call id");
+    let result = setup
+        .env
+        .await_call(call_id)
+        .expect("the fetch request didn't finish");
+    match &result {
+        WasmResult::Reply(bytes) => {
+            let response = Decode!(bytes, CheckTransactionResult).unwrap();
+            assert!(matches!(response, Err(CheckTransactionError::Txid { .. })));
+        }
+        WasmResult::Reject(msg) => panic!("unexpected reject: {}", msg),
```

Comment:
nit: The part that transform `result` into a `CheckTransactionResult` (without the assert) I would extract to a separate method, since this will needed for more or less every test

Reply:
Thanks for the suggestion! Extracted to a `decode` function.